types.ts

```diff
@@ -0,0 +1,16 @@
+export interface Storage<T> {
+  put(record: T): void;
+  get(start?: number, end?: number): T[];
+}
```

Comment:
This in src/storage/base.ts then an src/index.ts to export all the exposable resources